internal/cli/cmd/cluster/buildkit.go

```diff
@@ -189,19 +237,22 @@ func parseInjectWorkerInfo(workerInfoFile string, requiredPlatform api.BuildPlat
 	return f, nil
 }
 
-func startBackgroundProxy(ctx context.Context, md buildxInstanceMetadata, connect bool, debugFile string, useGrpcProxy bool, staticWorkerDefFile string) (int, error) {
+func startBackgroundProxy(ctx context.Context, md buildxInstanceMetadata, connect bool, useGrpcProxy bool, staticWorkerDefFile string) (int, string, error) {
 	if connect {
 		// Make sure the cluster exists before going to the background.
 		if _, err := ensureBuildCluster(ctx, md.Platform); err != nil {
-			return 0, err
+			return 0, "", err
 		}
 	}
 
-	cmd := exec.Command(os.Args[0], "buildkit", "proxy", "--sock_path="+md.SocketPath, "--platform="+string(md.Platform), "--region="+api.RegionName)
-	if debugFile != "" {
-		cmd.Args = append(cmd.Args, "--debug_to_file="+debugFile)
+	cmd := exec.Command(os.Args[0], "buildkit", "proxy", "--sock_path="+md.SocketPath, "--platform="+string(md.Platform), "--region="+api.RegionName, "--child=true")
```

Comment:
So I think we should use a different design here:
- Have the parent figure out a stable directory to host control sockets.
- Pass an absolute path to the socket that the child should listen on.
- Don't need any additional signaling from the child back to the parent.
- Use Unix sockets to run the HTTP server instead of TCP.